python/hail/api1/context.py

```diff
@@ -872,11 +884,14 @@ def import_matrix(self, path, min_partitions=None, drop_samples=False, cell_type
             cell_type = TInt64()
         return VariantDataset(self,
                               self._jhc.importMatrices(jindexed_seq_args(path),
+                                                       number_of_annotations,
```

Comment:
Can you check annotation_header has the same number of fields as annotation_types if it is present?